lte/gateway/c/session_manager/ShardTracker.cpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <vector>
+
+#include "ShardTracker.h"
+
+namespace magma {
+
+int ShardTracker::add_ue() {
+  int shard_id;
+  // If there are no shards, add an entry for the shard
+  // representing a single UE at the new shard
+  if (ue_count_per_shard_.size() == 0) {
+    ue_count_per_shard_.push_back(1);
+    return 0;
+  } else {
+    // Iterate through all existing shards, if all are full
+    // create a new shard with quantity 1, otherwise increment
+    // the quantity of UEs in the latest shard
+    for (auto i = 0; i < ue_count_per_shard_.size(); i++) {
```

Comment:
Since the index refers to ShardID in this implementation, let's use shard_id instead of i.
```suggestion
    for (auto shard_id = 0; shard_id < ue_count_per_shard_.size(); shard_id++) {
```